components/web_discovery/browser/rsa.cc

```diff
@@ -9,43 +9,34 @@
 #include <vector>
 
 #include "base/base64.h"
-#include "base/check.h"
 #include "base/threading/thread_restrictions.h"
 #include "crypto/keypair.h"
-#include "crypto/rsa_private_key.h"
 #include "crypto/sign.h"
 
 namespace web_discovery {
 
-namespace {
-
-constexpr size_t kRsaKeySize = 2048;
-
-}  // namespace
-
 EncodedRSAKeyPair::EncodedRSAKeyPair(std::string private_key_b64,
                                      std::string public_key_b64)
     : private_key_b64(std::move(private_key_b64)),
       public_key_b64(std::move(public_key_b64)) {}
 EncodedRSAKeyPair::~EncodedRSAKeyPair() = default;
 
-ImportedRSAKey::ImportedRSAKey() = default;
+ImportedRSAKey::ImportedRSAKey(crypto::keypair::PrivateKey private_key,
+                               std::string public_key_b64)
+    : private_key(std::move(private_key)),
+      public_key_b64(std::move(public_key_b64)) {}
 ImportedRSAKey::~ImportedRSAKey() = default;
 
-std::unique_ptr<crypto::RSAPrivateKey> GenerateRSAKey() {
+crypto::keypair::PrivateKey GenerateRSAKey() {
   base::AssertLongCPUWorkAllowed();
-  return crypto::RSAPrivateKey::Create(kRsaKeySize);
+  return crypto::keypair::PrivateKey::GenerateRsa2048();
 }
 
 std::unique_ptr<EncodedRSAKeyPair> ExportRSAKey(
-    const crypto::RSAPrivateKey& private_key) {
-  std::vector<uint8_t> encoded_public_key;
-  std::vector<uint8_t> encoded_private_key;
-
-  if (!private_key.ExportPrivateKey(&encoded_private_key) ||
-      !private_key.ExportPublicKey(&encoded_public_key)) {
-    return nullptr;
-  }
+    const crypto::keypair::PrivateKey& private_key) {
+  std::vector<uint8_t> encoded_private_key = private_key.ToPrivateKeyInfo();
+  std::vector<uint8_t> encoded_public_key =
+      private_key.ToSubjectPublicKeyInfo();
```

Comment:
It looks like we don't need `EncodedRSAKeyPair` to begin with, and we should be using `crypto::keypair::PrivateKey` directly. Am I missing something else here?